src/backend/commands/player/toxic/kick_from_vehicle.cpp

```diff
@@ -15,15 +15,36 @@ namespace big
 
 		virtual void execute(player_ptr player, const command_arguments& _args, const std::shared_ptr<command_context> ctx) override
 		{
-			auto vehicle = player->get_current_vehicle();
+			int vehicle = PED::GET_VEHICLE_PED_IS_IN(PLAYER::GET_PLAYER_PED(player->id()), false);
```

Comment:
Pedantic, but the type returned from the natives is Vehicle not int. (We typedef it as int, because it takes up the same space in the C++ layer, but to the native VM operator it is different)

Reply:
I tested the trigger for all functions and finally came up with this, here are the codes:

```
//Trigger Function:

void func_1846(int iParam0, int iParam1, float fParam2, int iParam3, int iParam4, int iParam5, bool bParam6)//Position - 0xB0D90
{
	struct<10> Var0;
	
	Var0.f_0 = -503325966;
	Var0.f_1 = PLAYER::PLAYER_ID();
	Var0.f_8 = bParam6;
	Var0.f_5 = iParam1;
	Var0.f_6 = iParam4;
	Var0.f_3 = fParam2;
	Var0.f_4 = iParam3;
	Var0.f_7 = iParam5;
	Var0.f_9 = MISC::GET_FRAME_COUNT();
	if (!iParam0 == 0)
	{
		SCRIPT::_SEND_TU_SCRIPT_EVENT_NEW(1, &Var0, 10, iParam0, Var0.f_0);
	}
}
```

playermask = (int)iParam0

```
//Trigger Example:

void func_2358(int iParam0)//Position - 0xD0E9B
{
	bool bVar0;
	int iVar1;
	
	iVar1 = 0;
	while (iVar1 < 32)
	{
		bVar0 = PLAYER::INT_TO_PLAYERINDEX(iVar1);
		if (func_21(bVar0, 0, 1) && bVar0 != PLAYER::PLAYER_ID())
		{
			if (PED::IS_PED_IN_VEHICLE(PLAYER::GET_PLAYER_PED(bVar0), iParam0, false) && !func_2359(bVar0))
			{
				func_1846(func_24279(bVar0), 0, 0f, 0, 0, 0, -1);
				VEHICLE::SET_VEHICLE_DOORS_LOCKED_FOR_PLAYER(iParam0, bVar0, true);
			}
		}
		iVar1++;
	}
}
```

```
//Other Functions:

int func_134(var uParam0, int iParam1, bool bParam2)//Position - 0x6EEC
{
	if (iParam1 == -1)
	{
		return 1;
	}
	func_135(uParam0, bParam2, 0);
	if (NETWORK::NETWORK_IS_GAME_IN_PROGRESS() && !bParam2)
	{
		if (MISC::ABSI(NETWORK::GET_TIME_DIFFERENCE(NETWORK::GET_NETWORK_TIME(), *uParam0)) >= iParam1)
		{
			return 1;
		}
	}
	else if (MISC::ABSI(NETWORK::GET_TIME_DIFFERENCE(MISC::GET_GAME_TIMER(), *uParam0)) >= iParam1)
	{
		return 1;
	}
	return 0;
}

int func_47(bool bParam0)//Position - 0x1F11
{
	int iVar0;
	
	iVar0 = bParam0;
	if (iVar0 < 0)
	{
		return 0;
	}
	if (iVar0 >= 32)
	{
		return 0;
	}
	return 1;
}

int func_24279(bool bParam0)//Position - 0x705203
{
	var uVar0;
	
	if (func_47(bParam0))
	{
		MISC::SET_BIT(&uVar0, bParam0);
	}
	return uVar0;
}

int func_21(bool bParam0, bool bParam1, bool bParam2)//Position - 0x15C1
{
	int iVar0;
	
	iVar0 = bParam0;
	if (iVar0 != -1)
	{
		if (NETWORK::NETWORK_IS_PLAYER_ACTIVE(bParam0))
		{
			if (bParam1)
			{
				if (!PLAYER::IS_PLAYER_PLAYING(bParam0))
				{
					return 0;
				}
			}
			if (bParam2)
			{
				if (iVar0 == Global_2672855.f_3)
				{
					return Global_2672855.f_2;
				}
				else if (Global_2657971[iVar0 /*465*/] != 4)
				{
					return 0;
				}
			}
			return 1;
		}
	}
	return 0;
}
```